docker/README.md

```diff
@@ -0,0 +1,78 @@
+Docker environment for mender-convert
+=====================================
+
+In order to correctly set up partitions and bootloaders, mender-convert has many dependencies,
+and their version and name vary between Linux distributions.
+
+To make using mender-convert easier, a reference setup using a Ubuntu 18.04 Docker container
+is provided.
+
+You need to [install Docker Engine CE](https://docs.docker.com/install) to use this environment.
+
+
+## Build the mender-convert container image
+
+To build a container based on Ubuntu 18.04 with all required dependencies for mender-convert,
+copy this directory to your workstation and change the current directory to it.
+
+Then run
+
+```bash
+./build.sh
+```
+
+This will create a container image you can use to run mender-convert.
+
+
+## Use the mender-convert container image
+
+Create a directory `input` under the directory where you copied these files (`build.sh`, `run.sh`, etc.):
+
+```bash
+mkdir input
+```
+
+Put your raw disk image into `input`, e.g. `input/2018-10-09-raspbian-stretch.img`.
+Then put a Mender client binary compiled for your device into `input/mender-1.6.0-arm-linux-gnueabihf-gcc`.
+See [the Mender documentation on how to cross-compile the Mender client for your device](https://docs.mender.io/development/client-configuration/cross-compiling).
```

Comment:
Would be nice to add this to the container as well, either pre-built "mender" binary or the steps necessary to compile it.

Might be an optimization for later.

Reply:
@estenberg  In my opinion we should firstly perform manual conversion inside docker, and then try to add scripts to do it in automatic way. Adding initial support for docker without checking everything carefully can cost us a lot of effort without any significant results.

Reply:
@mirzak agreed, we need to make the mender client packaging/compilation easier. This was just the first step in many. :) I was also planning to
* add support for taking everything as arguments to the container (either ARG or ENV), so we don't have to work inside the container itself
* publish it on docker hub somewhere, so users don't have to build it
* get the right Mender client for the board, like you mention. This is the most "hairy" one, as it needs to match the architecture (at least ARM vs. x86). Any help here appreciated! :)